File: Languages/CSharp/templates/new/fixture.rb
#!/usr/bin/env ruby

if ARGV[0] == 'get_file_extension'
	puts '.cs'
	exit
end

if ARGV[0] == 'get_position'
	puts '11|4'
	exit
end

if ARGV[0] == 'get_definition'
	puts "Creates an empty NUnit test fixture for C#"
	exit
end

classname = ARGV[0]
namespace = ARGV[1]
parameterfile = ARGV[2]

puts "using System;
using NUnit.Framework;

namespace #{namespace}
{
	[TestFixture]
	public class #{classname}
	{
		[Test]
		public void Test()
		{
		}
	}
}"

File: Languages/CSharp/templates/new/fixture.py
#!/usr/bin/env python
import sys

if __name__ == "__main__":
    if sys.argv[1] == 'get_file_extension':
        print(".cs")
    elif sys.argv[1] == 'get_position':
        print("11|4")
    elif sys.argv[1] == 'get_definition':
        print("Creates an empty NUnit test fixture for C#")
    else:
        classname = sys.argv[1]
        namespace = sys.argv[2]
        parameterfile = sys.argv[3]
        print("using System;")
        print("using NUnit.Framework;")
        print("")
        print("namespace " + namespace)
        print("{")
        print(" [TestFixture]")
        print(" public class " + classname)
        print(" {")
        print("     [Test]")
        print("     public void Test()")
        print("     {")
        print("     }")
        print(" }")
        print("}")
